admin_guide/sdn_troubleshooting.adoc

```diff
@@ -0,0 +1,547 @@
+router machine may be reachable.  In which case, try using the telnet
+command to access the port for the router directly:
+....
+telnet 1.2.3.4 8000
+....
+
+You may get:
+....
+Trying 1.2.3.4...
+Connected to 1.2.3.4.
+Escape character is '^]'.
+....
+
+If so, there's something listening on the port on the IP address.
+That's good.  Hit `ctrl-]` then hit the _enter_ key and then type
+`close` to quit telnet.  Move on to the <<debugging-the-router>>
+section to check other things on the router.
+
+Or you could get:
+....
+Trying 1.2.3.4...
+telnet: connect to address 1.2.3.4: Connection refused
+....
+
+Which tells us that the router is not listening on that port.  Please
+see the <<debugging-the-router>> section for more pointers on how to
+configure the router.
+
+
+Or if you see:
+[source,sh]
+....
+Trying 1.2.3.4...
+  telnet: connect to address 1.2.3.4: Connection timed out
+....
+
+Which tells us that you can't talk to anything on that IP address.
+Check your routing, firewalls, and that you have a router listening on
+that IP address.  To debug the router, see the
+<<debugging-the-router>> section. For IP routing and firewall issues,
+debugging that is beyond the purview of this guide.
+
+
+== Debugging the Router
+
+Now that you have an IP address, we need to `ssh` to that machine and
+check that the router software is running on that machine and
+configured correctly.  So let's `ssh` there and make sure the router
+is running:
+....
+# oc get endpoints --namespace=default --label=router
+NAMESPACE   NAME              ENDPOINTS
+default     router            10.1.0.4:80
+....
+
+If that command fails, then your OpenShift configuration is broken.
+Fixing that is outside the scope of this document.
+
+You should see one or more router endpoints listed, but that won't
+tell you if they are running on the machine with the given external IP
+address, since the endpoint IP address will be one of the pod
+addresses that is internal to the cluster.  To get the list of router
+host IP addresses, run:
+....
+# oc get pods --all-namespaces --selector=router --template='{{range .items}}HostIP: {{.status.hostIP}}   PodIP: {{.status.podIP}}{{end}}{{"\n"}}'
+HostIP: 192.168.122.202   PodIP: 10.1.0.4
+....
+
+You should see the host IP that corresponds to your external address.
+If you do not, please refer to the
+link:architecture/core_concepts/routes.html[router documentation] to
+configure the router pod to run on the right node (by setting the
+affinity correctly) or update your DNS to match the IP addresses where
+the routers are running.
+
+At this point in the guide, you should be on a node, running your
+router pod, but you still can not get the http request to work.  First
+we need to make sure that the router is mapping the external URL to
+the correct service, and if that works, we need to dig into that
+service to make sure that all endpoints are reachable.
+
+Let's list all of the routes that OpenShift knows about:
```

Comment:
The router is, but routes themselves can be created in any project, namespaced to that project. Also the way it mentions "list all of the routes that OpenShift knows about" made me expect to see `--all-namespaces`.

For example, here's an environment (running `oc` as **system:admin**) with routes in various projects (note I happen to not have any in the default namespace):

https://gist.githubusercontent.com/adellape/d841fdae82ebd05299f7/raw/c033ffb6f63225f7a1edb7b8703cddd876de8f3d/gistfile1.txt